cardano-api/src/Cardano/Api/SerialiseTextEnvelope.hs

```diff
@@ -191,11 +191,15 @@ serialiseToTextEnvelope :: forall a. HasTextEnvelope a
                         => Maybe TextEnvelopeDescr -> a -> TextEnvelope
 serialiseToTextEnvelope mbDescr a =
     TextEnvelope {
-      teType    = textEnvelopeType ttoken
+      teType    = teType $ textEnvelopeType ttoken
     , teDescription   = fromMaybe (textEnvelopeDefaultDescr a) mbDescr
     , teRawCBOR = serialiseToCBOR a
     }
   where
+    teType :: [TextEnvelopeType] -> TextEnvelopeType
+    teType [] = ""
+    teType (x : _) = x
```

Comment:
Paraphrasing, the `teType` is the first in the list or the empty string.  If there original list had more than one, what is the significance of the ordering?

Reply:
In the exiting code there is always one value.
This implementation changes the type to use a list.
I don't think that is the right design.
At the very least it could use a a `NonEmpty`
or change the class to `primary_type` + `alternative_input_types`.

Reply:
It will default to the first in the list.  When reading a `TextEnvelope` file, it will check for all the types in the list hence preserving backwards compatibility. However when producing a `TextEnvelope` we simply default to the first in the list if there is more than one `teType`.